ttnn/cpp/ttnn/graph_processor.cpp

```diff
@@ -0,0 +1,450 @@
+#include "graph_processor.hpp"
+#include "tt_metal/tt_stl/reflection.hpp"
+#include "types.hpp"
+
+
+#include <cxxabi.h>
+#include <memory>
+#include <typeindex>
+#include <unordered_map>
+
+namespace {
+const std::string demangle(const char* name) {
+
+    int status = -4;
+
+    char* res = abi::__cxa_demangle(name, NULL, NULL, &status);
+
+    const char* const demangled_name = (status==0)?res:name;
+
+    std::string ret_val(demangled_name);
+
+    free(res);
+
+    return ret_val;
+}
+
+std::string tensorMemoryLayoutToString(TensorMemoryLayout layout) {
+    switch (layout) {
+        case TensorMemoryLayout::INTERLEAVED:
+            return "INTERLEAVED";
+        case TensorMemoryLayout::SINGLE_BANK:
+            return "SINGLE_BANK";
+        case TensorMemoryLayout::HEIGHT_SHARDED:
+            return "HEIGHT_SHARDED";
+        case TensorMemoryLayout::WIDTH_SHARDED:
+            return "WIDTH_SHARDED";
+        case TensorMemoryLayout::BLOCK_SHARDED:
+            return "BLOCK_SHARDED";
+        default:
+            return "UNKNOWN"; // Handle unexpected values
+    }
+}
+
+template<class V>
+std::type_info const& var_type(V const& v){
```

Comment:
I'd rename `V` to `VariantT` and `var_type` to `variant_typeid` just for clarity.

Reply:
done